airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -221,6 +225,7 @@ def ti_run(
             # TODO: Add variables and connections that are needed (and has perms) for the task
             variables=[],
             connections=[],
+            xcom_keys_to_clear=xcom_keys,
```

Comment:
Why are we sending the keys to be clearer to the client?

Shouldn't we be just deleting those keys in the server here?